Mathlib/AlgebraicGeometry/Noetherian.lean

```diff
@@ -0,0 +1,350 @@
+/-
+Copyright (c) 2024 Geno Racklin Asher. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Geno Racklin Asher
+-/
+import Mathlib.AlgebraicGeometry.AffineScheme
+import Mathlib.AlgebraicGeometry.Morphisms.QuasiCompact
+import Mathlib.AlgebraicGeometry.Morphisms.QuasiSeparated
+import Mathlib.AlgebraicGeometry.PrimeSpectrum.Noetherian
+import Mathlib.AlgebraicGeometry.Scheme
+import Mathlib.RingTheory.Localization.Submodule
+import Mathlib.Topology.NoetherianSpace
+import Mathlib.Topology.QuasiSeparated
+
+/-!
+# Noetherian and Locally Noetherian Schemes
+
+We introduce the concept of (locally) Noetherian schemes,
+giving definitions, equivalent conditions, and basic properties.
+
+## Main definitions
+
+* `AlgebraicGeometry.IsLocallyNoetherian`: a scheme is locally Noetherian
+  if the components of the structure sheaf at each affine open are Noetherian rings.
+
+* `AlgebraicGeometry.IsNoetherian`: a scheme is Noetherian if it is locally Noetherian
+  and compact as a topological space.
+
+## Main results
+
+* `AlgebraicGeometry.isLocallyNoetherian_iff_affine_cover`: a scheme is locally Noetherian
+  if and only if it is covered by affine opens whose sections are Noetherian rings.
+
+* `AlgebraicGeometry.IsLocallyNoetherian.quasiSeparated`: a locally Noetherian scheme is
+  quasi-separated.
+
+* `AlgebraicGeometry.isNoetherian_iff_finite_affine_cover`: a scheme is Noetherian
+  if and only if it is covered by finitely many affine opens whose sections are Noetherian rings.
+
+* `AlgebraicGeometry.IsNoetherian.noetherianSpace`: a Noetherian scheme is
+  topologically a Noetherian space.
+
+## References
+
+* [Stacks: Noetherian Schemes](https://stacks.math.columbia.edu/tag/01OU)
+* [Robin Hartshorne, *Algebraic Geometry*][Har77]
+
+-/
+
+universe u
+
+open Opposite AlgebraicGeometry Localization IsLocalization TopologicalSpace
+
+namespace AlgebraicGeometry
+
+/-- A scheme `X` is locally Noetherian if `𝒪ₓ(U)` is Noetherian for all affine `U`. -/
+class IsLocallyNoetherian (X : Scheme) : Prop where
+  component_noetherian : ∀ (U: X.affineOpens),
+    IsNoetherianRing (X.presheaf.obj (op U)) := by infer_instance
+
+section localizationProps
+
+variable {R : Type u} [CommRing R] (S : Finset R) (hS: Ideal.span (α := R) S = ⊤)
+  (hN: ∀ s : S, IsNoetherianRing (Away (M := R) s))
+
+lemma ideal_eq_iInf_away (I : Ideal R) :
```

Comment:
This does not belong here.

Reply:
I've moved it to `RingTheory.Localization.Ideal` which seems like a better spot for it.